src/ast/passes/codegen_llvm.cpp

```diff
@@ -2597,6 +2603,22 @@ ScopedExpr CodegenLLVM::visit(Cast &cast)
     auto *v = b_.CreateAllocaBPF(scoped_expr.value()->getType());
     b_.CreateStore(scoped_expr.value(), v);
     return ScopedExpr(v, [this, v] { b_.CreateLifetimeEnd(v); });
+  } else if (cast.cast_type.IsBoolTy()) {
+    if (cast.expr.type().IsStringTy()) {
+      if (cast.expr.type().GetSize() == 0) {
```

Comment:
Is this something that can happen?

Reply:
Probably not. Just being overly paranoid I guess. I can remove.